python_modules/libraries/dagster-dg/dagster_dg/generate.py

```diff
@@ -29,39 +32,82 @@ def generate_deployment(path: Path) -> None:
+PYPI_DAGSTER_DEPENDENCIES = ("dagster-components",)
+PYPI_DAGSTER_DEV_DEPENDENCIES = ("dagster-webserver",)
+
+
+def get_pyproject_toml_dependencies(use_editable_dagster: bool) -> str:
+    deps = EDITABLE_DAGSTER_DEPENDENCIES if use_editable_dagster else PYPI_DAGSTER_DEPENDENCIES
+    return "\n".join(
+        [
+            "dependencies = [",
+            *[f'    "{dep}",' for dep in deps],
+            "]",
+        ]
     )
 
-    editable_dagster_uv_sources = textwrap.dedent(f"""
-    [tool.uv.sources]
-    dagster = {{ path = "{editable_dagster_root}/python_modules/dagster", editable = true }}
-    dagster-graphql = {{ path = "{editable_dagster_root}/python_modules/dagster-graphql", editable = true }}
-    dagster-pipes = {{ path = "{editable_dagster_root}/python_modules/dagster-pipes", editable = true }}
-    dagster-webserver = {{ path = "{editable_dagster_root}/python_modules/dagster-webserver", editable = true }}
-    dagster-components = {{ path = "{editable_dagster_root}/python_modules/libraries/dagster-components", editable = true }}
-    dagster-embedded-elt = {{ path = "{editable_dagster_root}/python_modules/libraries/dagster-embedded-elt", editable = true }}
-    dagster-dbt = {{ path = "{editable_dagster_root}/python_modules/libraries/dagster-dbt", editable = true }}
+
+def get_pyproject_toml_dev_dependencies(use_editable_dagster: bool) -> str:
+    deps = (
+        EDITABLE_DAGSTER_DEV_DEPENDENCIES if use_editable_dagster else PYPI_DAGSTER_DEV_DEPENDENCIES
+    )
+    return "\n".join(
+        [
+            "dev = [",
+            *[f'    "{dep}",' for dep in deps],
+            "]",
+        ]
+    )
+
+
+def get_pyproject_toml_uv_sources(editable_dagster_root: str) -> str:
+    return textwrap.dedent(f"""
+        [tool.uv.sources]
+        dagster = {{ path = "{editable_dagster_root}/python_modules/dagster", editable = true }}
+        dagster-graphql = {{ path = "{editable_dagster_root}/python_modules/dagster-graphql", editable = true }}
+        dagster-pipes = {{ path = "{editable_dagster_root}/python_modules/dagster-pipes", editable = true }}
+        dagster-webserver = {{ path = "{editable_dagster_root}/python_modules/dagster-webserver", editable = true }}
+        dagster-components = {{ path = "{editable_dagster_root}/python_modules/libraries/dagster-components", editable = true }}
+        dagster-embedded-elt = {{ path = "{editable_dagster_root}/python_modules/libraries/dagster-embedded-elt", editable = true }}
+        dagster-dbt = {{ path = "{editable_dagster_root}/python_modules/libraries/dagster-dbt", editable = true }}
     """)
 
+
+def generate_code_location(path: Path, editable_dagster_root: Optional[str] = None) -> None:
+    click.echo(f"Creating a Dagster code location at {path}.")
+
     if editable_dagster_root:
-        uv_sources = editable_dagster_uv_sources
+        dependencies = get_pyproject_toml_dependencies(use_editable_dagster=True)
```

Comment:
`        dependencies = get_pyproject_toml_dependencies(use_editable_dagster=editable_dagster_root)`

and consolidate?